components/eventing-controller/reconciler/subscription/reconciler_test.go

```diff
@@ -89,71 +94,184 @@ var _ = Describe("Subscription Reconciliation Tests", func() {
 			givenSubscription.Spec.Sink = "invalid"
 			ensureSubscriptionCreated(givenSubscription, ctx)
 
 			By("Setting APIRule status to False")
-			subscriptionAPIReadyFalseCondition := eventingv1alpha1.MakeCondition(eventingv1alpha1.ConditionAPIRuleStatus, eventingv1alpha1.ConditionReasonAPIRuleStatusNotReady, v1.ConditionFalse, "")
+			subscriptionAPIReadyFalseCondition := eventingv1alpha1.MakeCondition(
+				eventingv1alpha1.ConditionAPIRuleStatus,
+				eventingv1alpha1.ConditionReasonAPIRuleStatusNotReady,
+				v1.ConditionFalse,
+				"",
+			)
 			getSubscription(givenSubscription, ctx).Should(And(
 				reconcilertesting.HaveSubscriptionName(subscriptionName),
 				reconcilertesting.HaveCondition(subscriptionAPIReadyFalseCondition),
 			))
 
-			By("Deleting the object to not provoke more reconciliation requests")
-			Expect(k8sClient.Delete(ctx, givenSubscription)).Should(BeNil())
-			getSubscription(givenSubscription, ctx).ShouldNot(reconcilertesting.HaveSubscriptionFinalizer(Finalizer))
+			By("Fixing the Subscription with a valid Sink")
+			path := "/path1"
+			validSink := fmt.Sprintf("https://%s.%s.svc.cluster.local%s", subscriberSvc.Name, subscriberSvc.Namespace, path)
+			givenSubscription.Spec.Sink = validSink
+			updateSubscription(givenSubscription, ctx).Should(reconcilertesting.HaveSubscriptionSink(validSink))
+
+			By("Creating a valid APIRule")
+			getAPIRuleForASvc(subscriberSvc, ctx).Should(reconcilertesting.HaveNotEmptyAPIRule())
+			apiRuleUpdated := filterAPIRulesForASvc(getAPIRules(ctx, subscriberSvc), subscriberSvc)
+			getAPIRule(&apiRuleUpdated, ctx).Should(And(
+				reconcilertesting.HaveNotEmptyHost(),
+				reconcilertesting.HaveNotEmptyAPIRule(),
+				reconcilertesting.HaveAPIRuleSpecRules(acceptableMethods, object.OAuthHandlerName, path),
+				reconcilertesting.HaveAPIRuleOwnersRefs(givenSubscription.UID),
+			))
+
+			By("Updating the APIRule status to be Ready")
+			ensureAPIRuleStatusUpdatedWithStatusReady(&apiRuleUpdated, ctx).Should(BeNil())
+
+			By("Setting a Subscription active condition")
+			subscriptionActiveCondition := eventingv1alpha1.MakeCondition(
+				eventingv1alpha1.ConditionSubscriptionActive,
+				eventingv1alpha1.ConditionReasonSubscriptionActive,
+				v1.ConditionTrue,
+				"",
+			)
+
+			By("Setting APIRule status in Subscription to Ready")
+			subscriptionAPIReadyCondition := eventingv1alpha1.MakeCondition(
+				eventingv1alpha1.ConditionAPIRuleStatus,
+				eventingv1alpha1.ConditionReasonAPIRuleStatusReady,
+				v1.ConditionTrue,
+				"",
+			)
+			getSubscription(givenSubscription, ctx).Should(And(
+				reconcilertesting.HaveSubscriptionName(subscriptionName),
+				reconcilertesting.HaveCondition(subscriptionActiveCondition),
+				reconcilertesting.HaveAPIRuleName(apiRuleUpdated.Name),
+				reconcilertesting.HaveCondition(subscriptionAPIReadyCondition),
+				reconcilertesting.HaveSubscriptionReady(),
+			))
+
+			By("Cleaning the application name and not sending invalid webhook requests")
```

Comment:
Why is this expectation needed in this test?

Reply:
The subscription is created with an unclean event type which consists of an unclean application name, like this: `"test.app_1-0+2=3"`

The application name should be cleaned and that's why there shouldn't be any invalid (=unclean) webhook requests

Would you prefer clarifying, that the cleaning step isn't happening here, but we're only checking here that it has happened?
Or rather send a clean event type in all the tests and add a new test which focuses on checking for the cleaned event type like:
`When: creating a subscription with an unclean event type
It: should clean the application name and send valid web hook requests only `